src/coreclr/vm/loaderallocator.cpp

```diff
@@ -1181,6 +1185,19 @@ void LoaderAllocator::Init(BaseDomain *pDomain, BYTE *pExecutableHeapMemory)
     m_pHighFrequencyHeap->m_fPermitStubsWithUnwindInfo = TRUE;
 #endif
 
+    if (dwStaticsHeapReserveSize != 0)
+    {
+        m_pStaticsHeap = new (&m_StaticsHeapInstance) LoaderHeap(STATIC_FIELD_HEAP_RESERVE_SIZE,
+                                                                 STATIC_FIELD_HEAP_COMMIT_SIZE,
+                                                                 initReservedMem,
+                                                                 dwStaticsHeapReserveSize);
+        initReservedMem += dwStaticsHeapReserveSize;
+    }
+    else
+    {
+        m_pStaticsHeap = m_pHighFrequencyHeap;
```

Comment:
```suggestion
        _ASSERTE(m_pHighFrequencyHeap != NULL);
        m_pStaticsHeap = m_pHighFrequencyHeap;
```

I assume this assert is appropriate. If not, why?

Reply:
Its not unreasonable. We have a requirement that the high frequency heap is always allocated, but an assert hurts nobody.